syft/pkg/cataloger/golang/parse_go_bin.go

```diff
@@ -65,23 +65,27 @@ func parseGoBin(location source.Location, reader io.ReadCloser, opener exeOpener
 func buildGoPkgInfo(location source.Location, mod, goVersion, arch string) []pkg.Package {
 	pkgsSlice := make([]pkg.Package, 0)
 	scanner := bufio.NewScanner(strings.NewReader(mod))
-
-	// filter mod dependencies: [dep, name, version, sha]
 	for scanner.Scan() {
 		fields := strings.Fields(scanner.Text())
 
-		// must have dep, name, version, sha
-		if len(fields) < 4 {
+		if len(fields) < 3 {
 			continue
 		}
 
-		if fields[0] == packageIdentifier || fields[0] == replaceIdentifier {
-			name := fields[1]
-			version := fields[2]
-			h1Digest := fields[3]
+		name := fields[1]
+		version := fields[2]
+		h1Digest := ""
+		if len(fields) >= 4 {
+			h1Digest = fields[3]
```

Comment:
nit: capturing the information about this case being for possible vendoring would be great --to help future devs in this file